manta-pay/abstract-protocol.tex

```diff
@@ -328,100 +350,140 @@ \subsection{Abstract Cryptographic Schemes}
+    Also, note that the $\ID$ value is optional. This is inhabited whenever there are $\sources$ or $\sinks$, but if the shape of the transaction is $(0, m, n, 0)$ then $\ID = \None$.
 \end{definition}
 
 In order for a \Transfer{} to be considered \emph{valid}, it must adhere to the following constraints:
 
 \begin{itemize}
+    \item \textbf{Correct Key Signing}: The keys used to construct \Sender{s} and \Receiver{s} are valid and can be signed by a unique \SpendingKey{}.
     \item \textbf{Same Id}: All the \AssetId{s} in the \Transfer{} must be equal.
     \item \textbf{Balanced}: The sum of input \AssetValue{s} must be equal to the sum of output \AssetValue{s}.
     \item \textbf{Well-formed Senders}: All of the \Sender{s} in the \Transfer{} must be constructed according to the above \Sender{} definition.
     \item \textbf{Well-formed Receivers}: All of the \Receiver{s} in the \Transfer{} must be constructed according to the above \Receiver{} definition.
 \end{itemize}
 
-In order to prove that these constraints are satisfied for a given \Transfer{}, we build a zero-knowledge proof which will witness that the \Transfer{} is valid and should be accepted by the \Ledger{}. It is not necessary to prove that the encryption of $\Receiver.\note$ and the decryption of a $\note$ from the \Ledger{} are valid. Deviation from the protocol in encryption or decryption stages does not reduce the security of the protocol for honest participants, it only makes certain assets inaccessible to honest receivers if they are not aware of the devation, since they cannot decrypt assets normally. This does not effect the \emph{balanced transfer} or \emph{ownership} invariants of the protocol for the existing assets of ledger participants.
+In order to prove that these constraints are satisfied for a given \Transfer{}, we build a zero-knowledge proof which will witness that the \Transfer{} is valid and should be accepted by the \Ledger{}.
 
 \begin{definition}[\Transfer{} Validity \Statement{}]\label{def:transfer-validity-statement}
     A transfer $T : \Transfer$ is considered \emph{valid} if and only if
     \begin{enumerate}
+        \item The signing authority is correctly constructed:
+            \begin{align*}
+                \ak_\alpha &:= \ak \cdot \alpha \\
+                \vk        &:= \KDF^{\vk}(\ak) \\
+                \pk        &:= g \cdot \vk
+            \end{align*}
         \item All the \AssetId{s} in $T$ are equal:
             \[
-                \abs{
-                    \left(\bigcup_{a \in T.\sources} a.\ID \right)
+                \abs{\,
+                    T.\ID
                     \cup
                     \left(\bigcup_{S \in T.\senders} S.\asset.\ID \right)
                     \cup
                     \left(\bigcup_{R \in T.\receivers} R.\asset.\ID \right)
-                    \cup
-                    \left(\bigcup_{a \in T.\sinks} a.\ID \right)
-                } = 1
+                \,} = 1
```

Comment:
do we really need to write this equation? The sentence above is shorter, clearer and unambiguous.

Reply:
Also, don't we need to add sources and sinks to the AssetId check?

Reply:
We have the single global optional `AssetId` for `Transfer`.